src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1401,12 +1402,29 @@ void LogCommand()
                 var p = s.InputParameters;
                 if (p.Count > 0 && (NpgsqlLogManager.IsParameterLoggingEnabled || connector.Settings.LogParameters))
                 {
-                    sb.Append('\t').Append("Parameters:");
                     for (var i = 0; i < p.Count; i++)
-                        sb.Append("\t$").Append(i + 1).Append(": ").Append(Convert.ToString(p[i].Value, CultureInfo.InvariantCulture));
+                    {
+                        sb.Append("\t").Append("Parameters $").Append(i + 1).Append(":");
+                        switch (p[i].Value)
+                        {
+                        case IList list:
+                            for (var j = 0; j < list.Count; j++)
+                            {
+                                sb.Append("\t#").Append(j).Append(": ").Append(Convert.ToString(list[j], CultureInfo.InvariantCulture));
```

Comment:
Nit: the \t can be extracted as it's common to all cases.

Reply:
Oh wow, i'm really not doing well on this, sorry. I'll sort this out now

Reply:
Note that at the end of the for loop there is a sb.AppendLine()